### PR TITLE
Async event when shard is downloaded

### PR DESCRIPTION
**Pull request by oleksiylukin**
Async message on shard download end

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShardDownloadPresenceObserver.java [hunk 1]
```diff
@@ -75,8 +76,21 @@ public ShardDownloadPresenceObserver(DatabaseManager databaseManager, Blockchain
      *
      * @param shardPresentData shard present data contains downloaded ZIP name
      */
-    public void onShardPresent(@Observes @ShardPresentEvent(ShardPresentEventType.SHARD_PRESENT) ShardPresentData shardPresentData) {
-        shardImporter.importShard(shardPresentData.getFileIdValue(), List.of());
+    public void onShardPresent(@ObservesAsync @ShardPresentEvent(ShardPresentEventType.SHARD_PRESENT) ShardPresentData shardPresentData) {
+        try {
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShardDownloadPresenceObserver.java]: Why we should wait here?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java [hunk 2]
```diff
@@ -99,6 +99,11 @@ public void importShard(String fileId, List<String> excludedTables) {
         log.debug("Try unpack file name '{}'", zipInFolder);
         boolean unpackResult = zipComponent.extract(zipInFolder.toString(), csvImporter.getDataExportPath().toString());
         log.debug("Zip is unpacked = {}", unpackResult);
+        if (!unpackResult) {
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java]: Looks like not the best solution

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShardDownloadPresenceObserver.java (hunk 1 above)
Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShardDownloadPresenceObserver.java]: It gives ability to unpack zip successfully. Looks like it works fine now for me locally.

Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShardDownloadPresenceObserver.java]: Need to find correct solution. Using Thread.sleep - is not acceptable method of synchronization, random aspect is too big

Comment (oleksiylukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShardDownloadPresenceObserver.java]: Already fixed

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java (hunk 2 above)
Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java]: Now it throws exception.

Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java]: Fixed to throw exception

**Pull request merged**